redbot/core/bank.py

```diff
@@ -669,3 +677,20 @@ def _invalid_amount(amount: int) -> bool:
         raise RuntimeError("Guild is missing and required.")
 
     return amount
+
+
+def cost(amount: int = 0) -> bool:
+    async def predicate(ctx):
+        try:
+            if await get_balance(ctx.author) >= amount:
+                await withdraw_credits(ctx.author, amount)
+                last_data = {"command": str(ctx.command.qualified_name), "cost": amount}
+                await _conf.member(ctx.author).last_command_cost.set(last_data)
+                return True
+            else:
+                await ctx.send("You don't have enough money to use that command !")
+                return False
+        except:
+            return True
+
+    return commands.check(predicate)
```

Comment:
This can't be a check decorator. This will subtract money for every use of help per usage of this decorator. Checks should not have interaction consequences beyond the current invocation being valid or not.